### PR TITLE
Add option to enable hue-mode as a deCONZ argument

### PR DESCRIPTION
**Pull request by KodeCR**
Without this command-line deconz argument option the basic config (without a user/api-key) isn't hue-compatible as the mode is normally based on the api-key. That seems to hinder initial hue-app pairing sometimes.

### CODE REVIEW

**Comment by ebaauw** [comment]
You also need to update `rest_configuration.cpp`, to provide the same values for an authenticated GET of `/config`.  Best define constants (in `de_web_plugin_private.h`) so future updates need to be made only in one place.

**Comment by KodeCR** [comment]
Thanks for reviewing. An authenticated GET /config uses configToMap, which calls basicConfigToMap first, which sets these variables, or did I miss something?

**Comment by ebaauw** [comment]
Ah, you moved generating these attributes to `basicConfigMap()`.  Sorry, I missed that.

**Comment by KodeCR** [comment]
No worries, thanks again for checking

File: rest_configuration.cpp [hunk 1]
```diff
@@ -1022,12 +1020,10 @@ void DeRestPluginPrivate::configToMap(const ApiRequest &req, QVariantMap &map)
     swupdate2["bridge"] = bridge;
     swupdate2["checkforupdate"] = false;
     swupdate2["state"] = gwSwUpdateState;
-    swupdate2["install"] = false;
     autoinstall["updatetime"] = "";
     autoinstall["on"] = false;
     swupdate2["autoinstall"] = autoinstall;
     swupdate2["lastchange"] = "";
-    swupdate2["lastinstall"] = "";
```
Comment (manup) [inline rest_configuration.cpp:1030]: "install" and "lastinstall" are present in the Hue API why remove them?

File: rest_configuration.cpp [hunk 2]
```diff
@@ -1079,19 +1075,30 @@ void DeRestPluginPrivate::configToMap(const ApiRequest &req, QVariantMap &map)
...
+        map["swversion"] = swversion;
+        map["apiversion"] = QString(GW_API_VERSION);
+        // map["modelid"] = QLatin1String("deCONZ");
+        map["datastoreversion"] = QLatin1String("93");
+    } 
+    else
+    {
+        map["swversion"] = QLatin1String("1941132070");
+        map["apiversion"] = QLatin1String("1.41.0");
+        // map["modelid"] = QLatin1String("BSB002");
```
Comment (manup) [inline rest_configuration.cpp:1095]: Since the Hue "modelid" was remove above it should to be set here I guess?

File: authorisation.cpp [hunk 3]
```diff
@@ -194,8 +194,13 @@ void DeRestPluginPrivate::authorise(ApiRequest &req, ApiResponse &rsp)
             {
                 // supports deCONZ specifics
             }
+            // else if (!(i->useragent.isEmpty()) && i->useragent.startsWith(QLatin1String("Mozilla"))) // Phoscon
+            // {
+            //     req.mode = ApiModeNormal;
+            // }
```
Comment (manup) [inline authorisation.cpp:200]: Please remove dead code from PR :)

**Review by manup — CHANGES_REQUESTED** [review]
Added a few comments, can you please have a look.

File: rest_configuration.cpp (hunk 1 above)
Comment (KodeCR) [reply rest_configuration.cpp:1030]: The Hue API documentation is not very clear on this, but from the actual bridge it looks like:
- "install" is a write only item
- "lastinstall" is only part of the swupdate2 state for the bridge subobject

File: rest_configuration.cpp (hunk 2 above)
Comment (KodeCR) [reply rest_configuration.cpp:1095]: "modelid" is set in line 1081 to "deCONZ", it seems that the Hue App at least doesn't care that it's not "BSB002".
I at least can remove the commented lines though...

File: authorisation.cpp (hunk 3 above)
Comment (KodeCR) [reply authorisation.cpp:200]: Sorry... will do :)
Thanks for reviewing

File: rest_configuration.cpp (hunk 2 above)
Comment (manup) [reply rest_configuration.cpp]: Good to know, the Hue App might be fine but have you tested this also with Amazon Echo?
If in doubt I'd vote to just keep the BSB002 `modelid`.

Comment (KodeCR) [reply rest_configuration.cpp]: I haven't tested it directly with the Echo, I'm using Alexa integration through homebridge. Just to be sure I've put back the BSB002 modelid.

**Comment by Smanar** [comment]
Hello, this feature is still working ?
Someone is using " --hue-mode" but he still have ApiMode = 0 in log.

**Pull request merged**